lab_2_retrieval_w_bm25/main.py

```diff
@@ -120,6 +175,25 @@ def calculate_bm25(
 
     In case of corrupt input arguments, None is returned.
     """
+    if (not vocab or not document or not idf_document or not isinstance(vocab, list) or
+            not isinstance(document, list)):
+        return None
+    if (not isinstance(idf_document, dict) or not isinstance(doc_len, int) or
+            not isinstance(k1, float) or not isinstance(b, float) or
+            not isinstance(avg_doc_len, float)):
+        return None
+    if (isinstance(doc_len, bool) or not all(isinstance(term, str) for term in vocab) or
+            not all(isinstance(term, str) for term in idf_document)):
+        return None
+    if (not all(isinstance(term, str) for term in document) or
+            not all(isinstance(freq, float) for freq in idf_document.values())):
+        return None
+    vocab.extend([term for term in document if term not in vocab])
+    idf_document.update({term: 0.0 for term in vocab if term not in idf_document})
+    return {term: (idf_document[term] * ((document.count(term) * (k1 + 1))
```

Comment:
looks massive, place the code more beautifully

Reply:
fixed (??)
hopefully